compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1240,7 +1239,32 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
             }
         }
+        // Only if an appropriate error source is not found, check method chain for possible candiates
+        let error_source_not_found = unsatisfied_predicates.is_empty();
+        if error_source_not_found && let Mode::MethodCall = mode && let SelfSource::MethodCall(mut rcvr_expr) = source {
+        while let hir::ExprKind::MethodCall(_path_segment, parent_expr, _args, method_span) =
+        rcvr_expr.kind
+            {
+                let prev_ty = self.resolve_vars_if_possible(
+                    self.typeck_results
+                        .borrow()
+                        .expr_ty_adjusted_opt(parent_expr)
+                        .unwrap_or(Ty::new_misc_error(self.tcx)),);
+
+                for _matched_method in self.probe_for_name_many(
+                    Mode::MethodCall,
+                    item_name,
+                    None,
+                    IsSuggestion(true),
+                    prev_ty,
+                    rcvr_expr.hir_id,
+                    ProbeScope::TraitsInScope,) {
+                        err.span_label(method_span, format!("{item_kind} `{item_name}` is available on `{prev_ty}`"));
```

Comment:
If you use `parent_expr.span`, it would make the error be correct, but when you have more than one element in the chain you would end up with overlapping spans which won't look great.

An option we have is to iterate once using the `while let` as you have it now, but collect the messages and spans in a vec and emit the diagnostics *afterwards* by iterating over the vec and *peeking* at the next span.